wfl-slack-app/src/handle-wfl-request.js

```diff
@@ -0,0 +1,154 @@
+var wflApi = require('./wfl-api')
+var slackApi = require('./slack-api')
+
+var Status = {
+    ACTIVE: 1,
+    DISCARDED: 2,
+    COMPLETED: 3
+}
+
+function getOrderDetails(bot, data, foodOrderJson) {
+    wflApi.getRestaurantById(foodOrderJson.properties.RestaurantId, function(restaurantJson) {
+        var message = 'Order Details:\n';
+        let restaurantName = restaurantJson.properties.Name;
+        message = message + 'Restaurant: ' + restaurantName + '\n';
+        if (foodOrderJson.entities) {
+            for (let selection of foodOrderJson.entities)
+            {
+                message = message + selection.properties.OrdererName + ': ' + selection.properties.Description + '\n';
+            }
+        }
+        bot.postMessage(data.channel, message);
+    });
+}
+
+function verifyAndSetStatus(bot, data, orderIdTable, status) {
+    verifyOrderIsActiveAndContinue(data, orderIdTable, function(activeOrderExists, foodOrderJson) {
+        if (activeOrderExists) {
+            wflApi.setFoodOrderStatus(orderIdTable[data.channel], status, function(responseJson) {
+                var actionCompleted = '';
+                if (status === Status.DISCARDED) {
+                    actionCompleted = 'discarded';
+                }
+                else if (status === Status.COMPLETED) {
+                    actionCompleted = 'completed';
+                }
+                bot.postMessage(data.channel, 'The order has been ' + actionCompleted + '.');
+            });
+        }
+        else {
+            bot.postMessage(data.channel, 'This channel does not have an active order.');
+        }
+    });
+}
+
+function createMenuSelection(bot, data, orderIdTable, menuSelectionIdTable, description) {
+    if(menuSelectionIdTable[data.user])
+    {
+        wflApi.deleteMenuSelection(orderIdTable[data.channel], menuSelectionIdTable[data.user], function(){});
+    }
+    return slackApi
+        .getUserInfo(data.user)
+        .then(name => wflApi.postMenuSelection(orderIdTable[data.channel], name, description, function(selectionJson) {
+            menuSelectionIdTable[data.user] = selectionJson.properties.Id;
+            var message = 'OK! ' + name + `'s` + ' selection is ' + description + '.';
+            bot.postMessage(data.channel, message);
+        }))
+}
+
+function deleteMenuSelection(bot, data, orderIdTable, menuSelectionIdTable) {
+    wflApi.deleteMenuSelection(orderIdTable[data.channel], menuSelectionIdTable[data.user], function() {
+        slackApi.getUserInfo(data.user)
+        .then(name => bot.postMessage(data.channel, name + `'s` + ' selection has been deleted.'))
+    });
+}
+
+function isNullOrWhitespace(input) {
+    if (typeof input === 'undefined' || input == null)
+        return true;
+    return input.replace(/\s/g, '').length < 1;
+}
+
+function verifyOrderIsActiveAndContinue(data, orderIdTable, callback) {
```

Comment:
While I don't think our main effort was learning javascript.  I think after this PR is merged it would make sense to discuss promises and how it can simplify some of this programming paradigm.  I know you used them a little bit with your fetch code.

Reply:
I have some limited experience with promises at the moment. Is there an advantage to using promises over passing callback functions into these methods? I liked being able to pass in various different callback functions to the functions accepting them.

Reply:
we should discuss promises a bit further at some point, just for awareness purposes.  There is a little bit of callback hell here, but as projects get more and more complex you tend to see more and more of these sorts of nesting callbacks.